kpet/data.py

```diff
@@ -451,6 +474,56 @@ def convert_tree_arches(self):
 
             self.trees[name]["arches"] = list(tree_arches)
 
+    def convert_sets(self):
+        """ Convert each suite / cases sets from a list of regexes
+        to a list of set names matching those regexes
+
+        Returns:
+            Raises a schema.Invalid exception when finding an invalid regex
+        """
+
+        unassigned = {'unassigned'}
+        error = ("One of '{0}: set:' regexes\n" +
+                 "doesn't match any of the available sets\n" +
+                 "The regex: '{1}'\n" +
+                 "The avaliable sets: " + str(list(self.sets.keys())))
+
+        # TODO: Remove after conversion
+        # This function is to convert from the old format (where sets were a
+        # part of the pattern)
+        def pattern_conversion(pattern):
```

Comment:
Please name functions in imperative mood, as a command. E.g. `convert_pattern`, not `pattern_conversion`.